ios/RCTFBSDK/core/RCTFBSDKProfile.m

```diff
@@ -0,0 +1,41 @@
+#import "RCTFBSDKProfile.h"
+#import <React/RCTLog.h>
+
+#import <React/RCTUtils.h>
+
+@implementation RCTFBSDKProfile
+
+RCT_EXPORT_MODULE(FBProfile);
+
+- (dispatch_queue_t)methodQueue
+{
+    return dispatch_get_main_queue();
+}
+
+RCT_EXPORT_METHOD(getCurrentProfile:(RCTResponseSenderBlock)callback)
+{
+    NSDictionary *profileDict = RCTBuildProfileDict();
+    callback(@[RCTNullIfNil(profileDict)]);
+}
+
+#pragma mark - Helper Functions
+
+static NSDictionary *RCTBuildProfileDict()
+{
+    if (!FBSDKProfile.currentProfile) {
+        return nil;
+    }
+
+    return @{
+        @"firstName": FBSDKProfile.currentProfile.firstName ? FBSDKProfile.currentProfile.firstName : [NSNull null],
+        @"lastName": FBSDKProfile.currentProfile.lastName ? FBSDKProfile.currentProfile.lastName : [NSNull null],
+        @"middleName": FBSDKProfile.currentProfile.middleName ? FBSDKProfile.currentProfile.middleName : [NSNull null],
+        @"imageURL": FBSDKProfile.currentProfile.imageURL ? FBSDKProfile.currentProfile.imageURL.relativeString : [NSNull null],
+        @"linkURL": FBSDKProfile.currentProfile.linkURL ? FBSDKProfile.currentProfile.linkURL.relativeString : [NSNull null],
+        @"userID": FBSDKProfile.currentProfile.userID ? FBSDKProfile.currentProfile.userID : [NSNull null],
+        @"email": FBSDKProfile.currentProfile.email ? FBSDKProfile.currentProfile.email : [NSNull null],
```

Comment:
On Android looks like we don't have this information coming from currentProfile, does iOS behave differently in such case? If yes, it would be nice to state it in the docs that only iOS should have value for this property.

Reply:
Good catch, I didn't catch this one

Reply:
I recognise now that even the iOS sdk retrieve the url with width and height (i.e. https://platform-lookaside.fbsbx.com/platform/profilepic/?asid=10224600543422918&height=100&width=100&....).
The iOS SDK set two default width and height values for the profile picture (100). In Android, we have only the method `getProfilePictureUri` with two params **mandatory**. So, I think that the correct way to handling this part is to pass 100 (as a default size setted by iOS) to the `getProfilePictureUri` in order to retrieve the same image for the two platforms. In the future (I think is out of scope of this PR), we can add a method that get the profile picture with the width and height parameters (even in iOS we have the possibility to make this). Could be the right way?

Reply:
Instead of having default values I would just stick with them as mandatory and later on when we implement the method to retrieve state it clearly.
And yes, I believe it can be part of a new PR, maybe you could please just create an issue and link it here to not lose track of this `getProfilePictureUri` method as someone else could pick it :)